File: backend/app/schemas/scenario_admin_permission.py
"""
场景管理员权限配置 Schema
"""

from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


class ScenarioAdminPermissionBase(BaseModel):
    """场景管理员权限配置基础 Schema"""
    scenario_basic_info: bool = True
    scenario_keywords: bool = True
    scenario_policies: bool = False
    playground: bool = True
    performance_test: bool = False


class ScenarioAdminPermissionCreate(ScenarioAdminPermissionBase):
    """创建场景管理员权限配置"""
    user_id: str
    scenario_id: str


class ScenarioAdminPermissionUpdate(BaseModel):
    """更新场景管理员权限配置"""
    scenario_basic_info: Optional[bool] = None
    scenario_keywords: Optional[bool] = None
    scenario_policies: Optional[bool] = None
    playground: Optional[bool] = None
    performance_test: Optional[bool] = None


class ScenarioAdminPermissionResponse(ScenarioAdminPermissionBase):
    """场景管理员权限配置响应"""
    id: str
    user_id: str
    scenario_id: str
    created_at: datetime
    updated_at: Optional[datetime] = None
    created_by: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

File: backend-java/src/main/java/com/llmguard/manager/domain/entity/ScenarioAdminPermission.java
package com.llmguard.manager.domain.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "scenario_admin_permissions")
public class ScenarioAdminPermission {

    @Id
    @Column(length = 36)
    private String id;

    @Column(name = "user_id", length = 36, nullable = false)
    private String userId;

    @Column(name = "scenario_id", length = 64, nullable = false)
    private String scenarioId;

    @Column(name = "scenario_basic_info", nullable = false)
    private Boolean scenarioBasicInfo;

    @Column(name = "scenario_keywords", nullable = false)
    private Boolean scenarioKeywords;

    @Column(name = "scenario_policies", nullable = false)
    private Boolean scenarioPolicies;

    @Column(nullable = false)
    private Boolean playground;

    @Column(name = "performance_test", nullable = false)
    private Boolean performanceTest;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "created_by", length = 36)
    private String createdBy;
}
